be/src/column/object_column.cpp

```diff
@@ -166,34 +166,46 @@ size_t ObjectColumn<T>::serialize_size() const {
 }
 
 template <typename T>
-uint8_t* ObjectColumn<T>::serialize_column(uint8_t* dst) {
-    encode_fixed32_le(dst, _pool.size());
-    dst += sizeof(uint32_t);
-
+bool ObjectColumn<T>::serialize_column(io::ZeroCopyOutputStream* out) {
+    io::CodedOutputStream os(out);
+    os.WriteLittleEndian32(static_cast<uint32_t>(_pool.size()));
     for (int i = 0; i < _pool.size(); ++i) {
-        uint64_t actual = _pool[i].serialize(dst + sizeof(uint64_t));
-        encode_fixed64_le(dst, actual);
-
-        dst += sizeof(uint64_t);
-        dst += actual;
+        auto size = _pool[i].serialize_size();
+        uint8_t* buffer = os.GetDirectBufferForNBytesAndAdvance(size);
+        if (buffer != nullptr) {
+            DCHECK_EQ(buffer + size, os.Cur());
+            uint64_t actual_size = _pool[i].serialize(buffer + sizeof(uint64_t));
+            buffer = io::CodedOutputStream ::WriteLittleEndian64ToArray(actual_size, buffer);
+            os.SetCur(buffer + actual_size);
+        } else {
+            // TODO: support serializing _pool[i] to io::CodedOutputStream directly
+            return false;
+        }
     }
-    return dst;
+    return !os.HadError();
 }
 
 template <typename T>
-const uint8_t* ObjectColumn<T>::deserialize_column(const uint8_t* src) {
-    uint32_t count = decode_fixed32_le(src);
-    src += sizeof(uint32_t);
-
+bool ObjectColumn<T>::deserialize_column(io::ZeroCopyInputStream* in) {
+    io::CodedInputStream is(in);
+    uint32_t count{};
+    if (!is.ReadLittleEndian32(&count)) return false;
     for (int i = 0; i < count; ++i) {
-        uint64_t size = decode_fixed64_le(src);
-        src += sizeof(uint64_t);
-
-        _pool.emplace_back(Slice(src, size));
-        src += size;
+        uint64_t size{};
+        if (!is.ReadLittleEndian64(&size)) return false;
+        if (UNLIKELY(size > INT_MAX)) return false;
```

Comment:
bitmap size could large than INT_MAX.

Reply:
> 

IMO, it's impossible in practice because our columns will only be serialized to protobuf now, and the protobuf has a hard memory limit of 2GB, which means you can never serialize a column greater than 2GB to protobuf.

Reply:
> 

But for late compatibility, I added these checks for safety.

I can fix this with additional checks if you think this is a practical problem.